src/main/java/com/minecolonies/coremod/colony/buildings/BuildingMiner.java

```diff
@@ -284,7 +286,9 @@ public AbstractJob createJob(final CitizenData citizen)
     @Override
     public boolean neededForWorker(@Nullable final ItemStack stack)
     {
-        return Utils.isPickaxe(stack) || Utils.isShovel(stack) || Utils.isAxe(stack);
+        return ItemStackUtils.hasToolLevel(stack, ToolType.PICKAXE, TOOL_LEVEL_WOOD_OR_GOLD, getMaxToolLevel())
```

Comment:
Hmm, True, better leave it this way then. We might want to separate the buildings in more abstract classes later on I think. AbstractBuildingStructureUser or similar, since the miner and builder seem to have a lot in common in the job and building classes as well.